### PR TITLE
LTS dense output

### PR DESCRIPTION
**Pull request by wthrowe**
### Types of changes:

- [ ] Bugfix
- [ ] New feature

### Component:

- [ ] Code
- [ ] Documentation
- [ ] Build system
- [ ] Continuous integration

### Code review checklist

- [ ] The PR passes all checks, including unit tests, `clang-tidy` and `IWYU`. For
  instructions on how to perform the CI checks locally refer to the [Dev guide
  on the Travis CI](https://spectre-code.org/travis_guide.html).
- [ ] The code is documented and the documentation renders correctly. Run `make doc`
  to generate the documentation locally into `BUILD_DIR/docs/html`. Then open
  `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).

### CODE REVIEW

File: src/Time/BoundaryHistory.hpp [hunk 1]
```diff
@@ -123,7 +123,8 @@ class BoundaryHistory {
   // We use pointers instead of iterators because deque invalidates
   // iterators when elements are inserted or removed at the ends, but
   // not pointers.
-  std::map<std::pair<const Time*, const Time*>, CouplingResult> coupling_cache_;
+  mutable std::map<std::pair<const Time*, const Time*>, CouplingResult>
```
Comment (kidder) [inline src/Time/BoundaryHistory.hpp:126]: are there any thread safety issues related to this map?

**Review by kidder — APPROVED** [review]

**Comment by kidder** [comment]
@nilsdeppe please look at

File: src/Time/BoundaryHistory.hpp (hunk 1 above)
Comment (nilsdeppe) [reply src/Time/BoundaryHistory.hpp:126]: I believe this goes in the DataBox, so no. @wthrowe is that correct?

**Review by nilsdeppe — APPROVED** [review]

File: src/Time/BoundaryHistory.hpp (hunk 1 above)
Comment (wthrowe) [reply src/Time/BoundaryHistory.hpp:126]: Right, this should never be accessed by multiple threads at once.

**Pull request merged**